src/server/src/main/java/io/cassandrareaper/service/SegmentRunner.java

```diff
@@ -311,6 +322,7 @@ private boolean runRepair() {
         } catch (InterruptedException e) {
           LOG.debug("Interrupted while sleeping after a segment was postponed... weird stuff...");
         }
+        releaseSegmentRunners();
```

Comment:
this won't be needed once https://github.com/thelastpickle/cassandra-reaper/pull/622/files#r284190582 is moved to be inside the `try {} finally {}` block.